src/modules/CaaSMapper.ts

```diff
@@ -113,13 +113,21 @@ export class CaaSMapper {
     return [identifier, this.locale].join('.')
   }
 
+  buildMediaUrl(url: string, rev?: number) {
+    if (rev && this.api.mode === FSXAContentMode.PREVIEW) {
+      url += `${url.includes('?') ? '&' : '?'}rev=${rev}`
```

Comment:
Depending on what kind of URLs will be passed into this function, this might break. A URL having an id at the end like `localhost:3000/route?p1=123#foo` would swallow the rev param.

Personally I'm not a fan of editing URLs manually in general, but would prefer to have a typed object or collection of utils for URLs that handle the editing on string level.
Might sound a little overkill for this, but could also be a good point to start a refactoring.

Reply:
As discussed with e-Spirit, _Media URLs_ never use anchors. But is possible that URLs start with `/` (instead of the origin), which makes URL parsing more difficult.